src/client/index.ts

```diff
@@ -276,12 +276,16 @@ export class Client {
 
 	async negotiatorConnectToWallet(walletType: string) {
 		let walletProvider = await this.getWalletProvider()
+		try {
+			let walletAddress = await walletProvider.connectWith(walletType)
 
-		let walletAddress = await walletProvider.connectWith(walletType)
-
-		logger(2, 'wallet address found: ' + walletAddress)
+			logger(2, 'wallet address found: ' + walletAddress)
 
-		return walletAddress
+			return walletAddress
```

Comment:
Can we remove all the try catches? It's better to let the exception propagate so when there are errors the message is specific and has the correct stack trace.

Reply:
done